### PR TITLE
Connections and pubsub peers cleanup

### PR DESCRIPTION
**Pull request by dryajov**

### CODE REVIEW

File: libp2p/protocols/pubsub/pubsub.nim [hunk 1]
```diff
@@ -87,11 +105,14 @@ proc sendSubs*(p: PubSub,
 
     await peer.sendSubOpts(topics, subscribe)
   except CancelledError as exc:
-    p.handleDisconnect(peer)
+    if not(isNil(peer)) and not(isNil(peer.conn)):
```
Comment (sinkingsugar) [inline libp2p/protocols/pubsub/pubsub.nim:108]: if peer is `nil` something is broken tho isn't it? Thought we fixed those issues

File: libp2p/protocols/pubsub/pubsub.nim [hunk 2]
```diff
@@ -266,6 +303,18 @@ proc sendHelper*(p: PubSub,
 
   return (published, failed)
 
+proc sendAndCleanup*(p: PubSub,
```
Comment (sinkingsugar) [inline libp2p/protocols/pubsub/pubsub.nim:306]: the name kind mislead me.. it gave me the illusion that the peer would be cleaned up all the time , maybe no need to mention the cleanup part.. dunno just sharing my feeling 😄

File: libp2p/protocols/pubsub/pubsub.nim (hunk 1 above)
Comment (dryajov) [reply libp2p/protocols/pubsub/pubsub.nim:108]: this is just a bit of defencive programming here.

File: libp2p/protocols/pubsub/pubsub.nim [hunk 3]
```diff
@@ -183,11 +219,9 @@ method unsubscribePeer*(p: PubSub, peerInfo: PeerInfo) {.base, async.} =
     let peer = p.peers[peerInfo.id]
 
     trace "unsubscribing from peer", peerId = $peerInfo
-    if not(isNil(peer.conn)):
+    if not(isNil(peer)) and not(isNil(peer.conn)):
```
Comment (dryajov) [inline libp2p/protocols/pubsub/pubsub.nim:222]: here however, as opposed to https://github.com/status-im/nim-libp2p/pull/279/files/378544033e73472db4761c527c4ba7bf7a5a0e0f#r456219385, can happen that the peer or connection are simply nil on arrival, which whould indicate an issue upstream, or the peer simply disconnected in the process of trying to establish a pubsub connection.

File: libp2p/protocols/pubsub/pubsub.nim (hunk 2 above)
Comment (dryajov) [reply libp2p/protocols/pubsub/pubsub.nim:306]: hmm maybe something like `publishAndDisconnect`, I wan't to make it explicit that this helper will cleanup failed peers/streams.

**Pull request merged**